src/field/arena.ts

```diff
@@ -189,30 +177,102 @@ export class Arena {
     return ret;
   }
 
+  /**
+   * Determines whether or not to reroll for the given
+   * @param pokemonSpecies
+   * @param level
+   *
+   * Mega/Primal/Ultra legendaries and Arceus (720) cannot spawn below level 90
+   * Most legendaries (including Regigigas but not Kyurem, Zacian and Zamazenta) cannot spawn below level 70
+   * All other sublegends cannot spawn below level 50
+   * The final base case only has Cosmoem/Cosmog
+   * @returns true if rerolling is required, false otherwise
+   */
+  determineRerollIfLegendLike(pokemonSpecies: PokemonSpecies, level: number): boolean {
+    if (pokemonSpecies.isLegendLike()) {
+      if (pokemonSpecies.baseTotal >= 720) {
+        return level < 90;
+      } else if (pokemonSpecies.baseTotal >= 670) {
+        return level < 70;
+      } else if (pokemonSpecies.baseTotal >= 580) {
+        return level < 50;
+      } else {
+        return level < 30;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Generates a boss BiomePoolTier for a given
```

Comment:
nit (same with other occurrences)
```suggestion
   * Generates a boss {@linkcode BiomePoolTier} for a given
```